src/browser/ui/dom/components/ReactDOMOption.js

```diff
@@ -86,7 +87,25 @@ var ReactDOMOption = ReactClass.createClass({
       props = assign({}, props, {selected: this.state.selected});
     }
 
-    return option(props, this.props.children);
+    var content = '';
+
+    // Flatten children and warn if they aren't strings or numbers;
+    // invalid types are ignored.
+    ReactChildren.forEach(this.props.children, function(child) {
+      if (child == null || typeof child === 'boolean') {
```

Comment:
You shouldn't need the boolean check here; forEach should map bools to null for you. Can you try removing this? I think your test will still pass.